internal/status/statefulset_status.go

```diff
@@ -68,52 +57,69 @@ func ResetStatefulSetStatus() {
 	utils.AviLog.Debugf("Successfully reset ako statefulset: %v", u)
 }
 
-// AddStatefulSetStatus sets a condition in status of AKO statefulset to the desired value
-func AddStatefulSetStatus(reason string, statusCondition v1.ConditionStatus) {
+func ResetStatefulSetAnnotation() {
 	ss, err := utils.GetInformers().ClientSet.AppsV1().StatefulSets(utils.GetAKONamespace()).Get(context.TODO(), lib.AKOStatefulSet, metav1.GetOptions{})
 	if err != nil {
 		utils.AviLog.Warnf("Error in getting ako statefulset: %v", err)
 		return
 	}
-
-	msg, ok := msgForReason[reason]
-	if !ok {
-		utils.AviLog.Warnf("Unknown reason %s for statefulset status", reason)
+	ann := ss.GetAnnotations()
+	if ann == nil {
+		return
+	}
+	if _, ok := ann[ObjectDeletionStatus]; !ok {
 		return
 	}
+	payloadValue := make(map[string]*string)
+	// To delete an annotation with patch call, the value has to be set to nil
+	payloadValue[ObjectDeletionStatus] = nil
 
-	if msgFoundInStatus(ss.Status.Conditions, msg) {
+	patchPayload := map[string]interface{}{
+		"metadata": map[string]map[string]*string{
+			"annotations": payloadValue,
+		},
+	}
+	payloadBytes, _ := json.Marshal(patchPayload)
+	_, err = utils.GetInformers().ClientSet.AppsV1().StatefulSets(utils.GetAKONamespace()).Patch(context.TODO(), ss.Name, types.MergePatchType, payloadBytes, metav1.PatchOptions{})
+
+	if err != nil {
+		utils.AviLog.Warnf("Error in patching ako statefulset: %v", err)
 		return
 	}
+	utils.AviLog.Infof("Successfully removed annotation %s from ako statefulset", ObjectDeletionStatus)
 
-	var foundCondition bool
-	currentTime := metav1.Now()
-	for i, c := range ss.Status.Conditions {
-		if c.Type == lib.AKOConditionType {
-			ss.Status.Conditions[i].Reason = reason
-			ss.Status.Conditions[i].Message = msg
-			ss.Status.Conditions[i].Status = statusCondition
-			ss.Status.Conditions[i].LastTransitionTime = currentTime
-			foundCondition = true
-			break
-		}
+	//Remove any status from previous versions of AKO
+	ResetStatefulSetStatus()
```

Comment:
is this required still?

Reply:
I thought that - if previous version of AKO had set the status value in statefulset, and then we move to a new version of AKO, then the status has to be cleaned up. 

Status in particular is not straightforward to cleanup using kubectl, so it may linger on. If thats fine, we can remove this call.

Reply:
will it create infinite loop ? ... ResetStatefulSetStatus --> ResetStatefulSetAnnotation and ResetStatefulSetAnnotation -->ResetStatefulSetStatus

Reply:
Great catch, removed ResetStatefulSetAnnotation call from ResetStatefulSetStatus